### PR TITLE
[WIP] Rabbit 5 snippets

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

File: Snippets/Rabbit/Rabbit_5/Usage.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    }
+
+    string MyRoutingKeyConvention(Type type)
+    {
+        throw new NotImplementedException();
+    }
+
+    void UseRoutingTopology5_0(EndpointConfiguration endpointConfiguration)
+    {
+        #region rabbitmq-config-useroutingtopologyDelegate 5.0
```
Comment (adamralph) [inline Snippets/Rabbit/Rabbit_5/Usage.cs]: Is the "5.0" required?

File: Snippets/Rabbit/Rabbit_5/Usage.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+        var transport = endpointConfiguration.UseTransport<RabbitMQTransport>();
+        transport.SetClientCertificates(new X509CertificateCollection());
+
+        #endregion
+    }
+
+    void DisableTimeoutManager(EndpointConfiguration endpointConfiguration)
+    {
+        #region rabbitmq-delay-disable-timeout-manager 5.0
```
Comment (adamralph) [inline Snippets/Rabbit/Rabbit_5/Usage.cs]: Is the "5.0" required?

**Comment by SimonCropp** [comment]
did the tests pass? eg https://github.com/Particular/docs.particular.net/blob/master/Snippets/Rabbit/Rabbit_All/QueueDeletion/Tests.cs

**Comment by adamralph** [comment]
@SimonCropp I didn't even know those existed. I'd love to run them now, but for the life of me I can't get the VS2017 15.3.2 test explorer to recognise them. Not even with `NUnit3TestAdapter` installed.

**Comment by adamralph** [comment]
```
[23.8.17 10:44:51 Informational] ------ Discover test started ------
[23.8.17 10:44:51 Informational] NUnit Adapter 3.8.0.0: Test discovery starting
[23.8.17 10:44:51 Informational] Assembly contains no NUnit 3.0 tests: C:\Code\Particular\docs.particular.net\Snippets\Common\bin\Debug\net462\Common.dll
[23.8.17 10:44:51 Informational] Assembly contains no NUnit 3.0 tests: C:\Code\Particular\docs.particular.net\Snippets\Rabbit\Rabbit_1\bin\Debug\net462\Rabbit_1.dll
[23.8.17 10:44:51 Informational] Assembly contains no NUnit 3.0 tests: C:\Code\Particular\docs.particular.net\Snippets\Rabbit\Rabbit_2\bin\Debug\net462\Rabbit_2.dll
[23.8.17 10:44:51 Informational] Assembly contains no NUnit 3.0 tests: C:\Code\Particular\docs.particular.net\Snippets\Rabbit\Rabbit_3\bin\Debug\net462\Rabbit_3.dll
[23.8.17 10:44:51 Informational] Assembly contains no NUnit 3.0 tests: C:\Code\Particular\docs.particular.net\Snippets\Rabbit\Rabbit_4\bin\Debug\net462\Rabbit_4.dll
[23.8.17 10:44:51 Informational] Assembly contains no NUnit 3.0 tests: C:\Code\Particular\docs.particular.net\Snippets\Rabbit\Rabbit_5\bin\Debug\net462\Rabbit_5.dll
[23.8.17 10:44:51 Warning] Exception System.OutOfMemoryException, Exception converting Rabbit_All.ErrorQueue.Tests.ReturnMessageToSourceQueue
[23.8.17 10:44:51 Warning] Exception of type 'System.OutOfMemoryException' was thrown.
[23.8.17 10:44:51 Warning] Exception System.OutOfMemoryException, Exception converting Rabbit_All.NativeSend.Tests.Send
[23.8.17 10:44:51 Warning] Exception of type 'System.OutOfMemoryException' was thrown.
[23.8.17 10:44:51 Warning] Exception System.OutOfMemoryException, Exception converting Rabbit_All.QueueCreation.Tests.CreateQueuesForEndpoint
[23.8.17 10:44:51 Warning] Exception of type 'System.OutOfMemoryException' was thrown.
[23.8.17 10:44:51 Warning] Exception System.OutOfMemoryException, Exception converting Rabbit_All.QueueDeletion.Tests.DeleteQueuesForEndpoint
[23.8.17 10:44:51 Warning] Exception of type 'System.OutOfMemoryException' was thrown.
[23.8.17 10:44:51 Informational] NUnit Adapter 3.8.0.0: Test discovery complete
[23.8.17 10:44:51 Informational] ========== Discover test finished: 0 found (0:00:00.5749996) ==========
```

File: Snippets/Rabbit/Rabbit_5/Usage.cs (hunk 1 above)
Comment (SimonCropp) [inline Snippets/Rabbit/Rabbit_5/Usage.cs]: nope

**Comment by SimonCropp** [comment]
try nunit 3.7

**Comment by adamralph** [comment]
It's a known bug. I found this workaround: https://github.com/nunit/nunit3-vs-adapter/issues/296#issuecomment-285698000

Two of the tests are failing. Investigating...

**Comment by adamralph** [comment]
Those two tests currently fail on master. I'll fix that first and then return here and rebase.

**Comment by adamralph** [comment]
Raised https://github.com/Particular/docs.particular.net/pull/3124 to fix the tests on master.

**Pull request merged**